### PR TITLE
HSEARCH-3006 Make the version of the Surefire Maven plugin a property

### PR DESCRIPTION
**Pull request by Sanne**
- https://hibernate.atlassian.net/browse/HSEARCH-3006
 - https://hibernate.atlassian.net/browse/HSEARCH-3007

### CODE REVIEW

**Comment by Sanne** [comment]
Wait a second, I'm going to send an improved version to actually use the right defaults with profiles.

**Comment by Sanne** [comment]
Updated.

File: pom.xml [hunk 1]
```diff
@@ -1878,20 +1878,44 @@ org.hibernate.search.spi.SearchIntegratorBuilder#buildSearchIntegrator()
...
+                <maven.plugin.failsafe.version>2.20.1</maven.plugin.failsafe.version>
+            </properties>
             <modules>
                 <module>documentation</module>
             </modules>
         </profile>
 
         <profile>
-            <id>jigsaw</id>
+            <id>jdk9</id>
```
Comment (yrodiere) [inline pom.xml]: Ok if I rename the "non-jigsaw" profile to jdk8 then?

File: pom.xml [hunk 2]
```diff
@@ -1878,20 +1878,44 @@ org.hibernate.search.spi.SearchIntegratorBuilder#buildSearchIntegrator()
                 <!-- Asciidoctor and Karaf won't work on JDK9 yet  -->
                 <jdk>1.8</jdk>
             </activation>
+            <properties>
+                <!-- Surefire versions are a minefield of bugs, so depending on context we need to choose a different one -->
+                <maven.plugin.surefire.version>2.20.1</maven.plugin.surefire.version>
+                <maven.plugin.failsafe.version>2.20.1</maven.plugin.failsafe.version>
```
Comment (yrodiere) [inline pom.xml]: Would you mind if I simply made this the default, and removed the overrides for jdk8 and jdk9? I'm afraid we'll forget to update both the next time we upgrade...

File: pom.xml (hunk 1 above)
Comment (Sanne) [reply pom.xml]: ok

File: pom.xml (hunk 2 above)
Comment (Sanne) [reply pom.xml]: ok. The reason I didn't set a default is to catch mistakes, I never trust its "inheritance rules" ;)

**Comment by yrodiere** [comment]
Merged, thanks!

**Pull request merged**